source/MRMesh/MRPointsComponents.cpp

```diff
@@ -103,14 +124,56 @@ Expected<std::vector<VertBitSet>> getLargestComponents( const PointCloud& pointC
     return result;
 }
 
+Expected<std::pair<std::vector<VertBitSet>, int>>  getAllComponents( const PointCloud& pointCloud, float maxDist, int maxComponentCount /*= 0*/, ProgressCallback pc /*= {} */ )
+{
+    MR_TIMER
+
+        assert( maxDist > 0.f );
+    const auto& validPoints = pointCloud.validPoints;
+    ProgressCallback subPc = subprogress( pc, 0.f, 0.9f );
+    auto unionStructsRes = getUnionFindStructureVerts( pointCloud, maxDist, nullptr, subPc );
+    if ( !unionStructsRes.has_value() )
+        return unexpectedOperationCanceled();
+    auto& unionStructs = *unionStructsRes;
+    const auto& allRoots = unionStructs.roots();
+
+    subPc = subprogress( pc, 0.9f, 0.95f );
+    auto [uniqueRootsMap, componentsCount] = getUniqueRootIds( allRoots, validPoints );
+    if ( !componentsCount )
+        return unexpected( std::string( "Error: no components found." ) );
```

Comment:
We do not add prefix "Error:" in other places

Reply:
fixed